utils/v2/connector/impl/fetchers/Fetcher.ts

```diff
@@ -0,0 +1,34 @@
+import { timeout } from "promise-timeout";
+
+export type SourceType = "cache-layer" | "streamr" | "streamr-storage";
+
+export interface SignedDataPackageResponse {
+  timestamp: number;
+  prices: { symbol: string; value: any }[];
+  signature: string;
+  liteSignature: string;
+}
+
+export interface SourceConfig {
+  type: SourceType;
+  url?: string; // required for "cache-layer" sources
+  streamrEndpointPrefix?: string; // required for "streamr" and "streamr-historical" sources
+  disabledForSinglePrices?: boolean;
+}
+
+export abstract class Fetcher {
+  constructor(protected config: SourceConfig, protected asset?: string) {}
```

Comment:
If the asset is not defined, we fetch the whole package, right?
Maybe we should comment on it, so it's easier to understand.

Reply:
Right. Great idea!

Reply:
Added the comment